table-structure.sql

```diff
@@ -0,0 +1,59 @@
+--User: id (pk), username, email, password
+--Role: id (pk), name
+--UserRoles: id(pk), user_id (fk), role_id (fk)   //A user may have multiple roles [Unique constraint on user_id, role_id]
+--Convention: id (pk), name, year   //startDate?, endDate?
+--Package: id (pk), name     //More fields for the "package" bought? ie include_shirt, num_of_days, etc?
+--Attendee: id (pk), user_id (fk), convention_id (fk), package_id (fk) [unique constraint on user_id, convention_id, package_id]
+
+
+
+CREATE TABLE user (
+  id INT AUTO_INCREMENT NOT NULL,
+  username VARCHAR(20) NOT NULL,
+  email VARCHAR(50) NOT NULL,
+  password VARCHAR(100) NOT NULL,
```

Comment:
Salty `bcrypt2` on this guy?

Reply:
@markmdavey Any guidance / resources around best practices around this would be fantastic. I, personally, do not have much experience surrounding user authentication.